hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimelineUtils.java

```diff
@@ -308,6 +309,38 @@ public static HoodieTimeline getCommitsTimelineAfter(
     return timelineSinceLastSync;
   }
 
+  public static HoodieTimeline getCommitsTimeLineAfterByCompletionTimeRange(
```

Comment:
If the completion time to resume from for a incremental query corresponds to an instant in the archived timeline, we should fallback to the snapshot query instead of pulling the archived timeline, as reading archived timeline can be expensive.

Reply:
Also, the data files added by the commits in the archived timeline may have already been deleted by the cleaner, so incremental query may not be able to read the files, and has to go through full scan approach.

Reply:
Fixed in this PR, and filed JIRA to track this issue for Hudi 0.x: https://issues.apache.org/jira/browse/HUDI-8357